cpp/terrain/terrain_object.cpp

```diff
@@ -184,17 +209,16 @@ void TerrainObject::set_ground(int id, int additional) {
 	}
 }
 
-void TerrainObject::annex(TerrainObject *other) {
-	this->children.push_back(other);
-	other->parent = this;
-}
-
 const TerrainObject *TerrainObject::get_parent() const {
 	return this->parent;
 }
 
 std::vector<TerrainObject *> TerrainObject::get_children() const {
-	return this->children;
+	std::vector<TerrainObject *> result;
+	for (auto &obj: this->children) {
+		result.push_back(obj.get());
+	}
+	return result;
```

Comment:
@Jon0: That's what I was thinking about, too.

The problem is that this does not really seem to be an established pattern in C/C++; I wonder why. After all, C/C++ usually are quite nazi when it comes to performance.